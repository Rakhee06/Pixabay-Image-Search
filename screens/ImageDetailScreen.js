import React, { Component } from 'react';
import {View,
    StyleSheet,
    ActivityIndicator,
    Linking,
    ScrollView
} from 'react-native';
import { Image,
    Icon,
    Card,
    ListItem
} from 'react-native-elements';
import { MonoText } from '../components/StyledText';

export default class ImageDetailScreen extends Component {
    state = {
        imageName: this.props.navigation.state.params.name,
        imageTags: this.props.navigation.state.params.tags,
        imageLikes: this.props.navigation.state.params.likes,
        imageLargeURL: this.props.navigation.state.params.imageURL,
        imageViews: this.props.navigation.state.params.views,
    };

    render(){
        const list = [
            {   icon: 'user',
                title: this.state.imageName
            }, {
                icon: 'eye',
                title: this.state.imageViews
            }, {
                icon: 'like1',
                title: this.state.imageLikes
            }, {
                icon: 'tags',
                title: this.state.imageTags
            }
        ];
        return (
            <ScrollView>
                <View style={styles.container}>
                    <Image
                        source={{ uri: this.state.imageLargeURL }}
                        style={{ width: 350, height: 350}}
                        PlaceholderContent={<ActivityIndicator
                            size="small"
                            color='#478692'
                        />}
                    />
                    <Card title='DETAILS'
                          containerStyle={{ marginRight: 25}}
                    >
                        {
                            list.map((u, i) => {
                                return(
                                    <ListItem key={i}
                                              subtitle={
                                                  <View style={styles.subContainer}>
                                                      <Icon
                                                          name={u.icon}
                                                          type='antdesign'
                                                          color='#478692'
                                                      />
                                                      <MonoText>{u.title}</MonoText>
                                                  </View>
                                              }
                                              bottomDivider={true}
                                    >
                                    </ListItem>
                                );
                            })
                        }
                    </Card>
                    <View
                        style={{ flex: 1, alignItems: 'center', justifyContent: 'center', marginTop: 15, marginBottom: 30 }}>
                        <MonoText
                            style={{ color: '#478692' }}
                            onPress={() => Linking.openURL('https://pixabay.com')}>
                            Pixabay Images
                        </MonoText>
                    </View>
                </View>
            </ScrollView>
        );
    };
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        marginLeft: 13,
        marginTop: 13
    },
    subContainer: {
        flexDirection: 'row',
        flexWrap: 'wrap',
        alignItems: 'center',
        justifyContent: 'space-between'
    }
});
