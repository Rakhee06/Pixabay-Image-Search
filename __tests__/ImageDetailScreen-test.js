import 'react-native';
import React from 'react';
import ImageDetailScreen from '../screens/ImageDetailScreen';
import renderer from 'react-test-renderer';

describe("Image Detail Screen", () => {
    it("should render image detail screen correctly", async () => {
        const navigationMock = { state: { params: {
                    name: 'qwe',
                    tags: 'sun, hot',
                    likes: '123',
                    imageURL: 'http://foo.bar',
                    views: '45'
        }} };

        const tree = await renderer.create(<ImageDetailScreen navigation={navigationMock}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});