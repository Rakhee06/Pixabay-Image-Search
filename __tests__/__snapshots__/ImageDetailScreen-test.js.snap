// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Detail Screen should render image detail screen correctly 1`] = `null`;
